cors/cors.go

```diff
@@ -0,0 +1,127 @@
+package cors
+
+import (
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+type OriginChecker interface {
+	Options(events.APIGatewayV2HTTPRequest) events.APIGatewayV2HTTPResponse
+	GetCorsHeaders(request events.APIGatewayV2HTTPRequest) (map[string]string, bool)
+}
+
+// Domains contains the allowed domains and the accepted methods for each one
+type Domains struct {
+	Allowed map[string]map[string]bool
+}
+
+// NewDomains returns the default Domains
+func NewOriginChecker() OriginChecker {
+	acceptedDomains := map[string]map[string]bool{
+		"thelist.app": {"DELETE": true, "GET": true, "PATCH": true, "POST": true},
+	}
+	return &Domains{Allowed: acceptedDomains}
+}
+
+func IsOptionsRequest(request events.APIGatewayV2HTTPRequest) bool {
+	return http.MethodOptions == request.RequestContext.HTTP.Method
+}
+
+// Options performs an options request
+// response details methods which are permitted to be performed from the domain in the Origin header
+func (d *Domains) Options(request events.APIGatewayV2HTTPRequest) events.APIGatewayV2HTTPResponse {
+	origin, ok := request.Headers["Origin"]
+	if !ok {
+		return events.APIGatewayV2HTTPResponse{
+			StatusCode: http.StatusBadRequest,
+			Body:       `{"error": "Origin header was not set on request"}`,
+		}
+	}
+
+	allowedMethds := d.getAllowedMethodsForOrigin(origin)
+	if len(allowedMethds) == 0 {
+		return events.APIGatewayV2HTTPResponse{
+			StatusCode: http.StatusNoContent,
+		}
+	}
+
+	return events.APIGatewayV2HTTPResponse{
+		StatusCode: http.StatusNoContent,
+		Headers: map[string]string{
+			"Access-Control-Allow-Origin":  origin,
+			"Access-Control-Allow-Methods": commaSeperateTrueKeys(allowedMethds),
+		},
+	}
+}
+
+func (d *Domains) GetCorsHeaders(request events.APIGatewayV2HTTPRequest) (map[string]string, bool) {
+	origin, ok := request.Headers["Origin"]
+	if !ok {
+		return nil, true
+	}
+
+	if !d.isOriginAllowedToPerformMethod(origin, request.RequestContext.HTTP.Method) {
+		return nil, false
```

Comment:
I'm not sure if this is the behaviour we want, if the origin isn't one we like we could still return the response, just without the `Access-Control-Allow-Origin` header?
```suggestion
		return map[string]string{}, true
```

Reply:
ohhhh - total missundertsanding. Will change signature of the function not thave a bool at all then